useragent/useragent.go

```diff
@@ -43,3 +43,18 @@ func UserAgent(binaryNameCapitalized string, version, commit, buildTime string,
 	builder.WriteByte(')')
 	return builder.String()
 }
+
+func UserAgentWithBeatTelemetry(binaryNameCapitalized string, version string, mode string, unprivileged string) string {
```

Comment:
```suggestion
func UserAgentWithBeatTelemetry(binaryNameCapitalized string, version string, mode AgentManagementMode, unprivileged bool) string {
```

The mode should probably be an enum so that it isn't totally arbitrary and unprivileged should be a bool. You can pre-emptively add FIPS as a bool in here if you want since we know we are going to need that.

Reply:
About making unprivileged a bool, I was thinking about it but found it less speaking from a reading perspective in the user-agent itself. 
Should we consider the parameter to be a boolean but have `<privileged | unprivileged>` written in the user-agent when we are building it? Same question goes for fips.

Reply:
The intent is to standardize the format of the user agent by eliminating the ability of the caller to control the exact text, only the components of the text. There is a size constraint to account for so we should centralize preserving the size constraint into this function and not expose it to callers at all. They shouldn't need to know it exists.

Instead of a bool you could use enums, which would probably make calls to the function easier to read.

Reply:
I added two enums. I let you confirm this is what you were expecting here.